v4/export/writer_util.go

```diff
@@ -114,6 +121,37 @@ func (b *writerPipe) ShouldSwitchStatement() bool {
 		(b.statementSizeLimit != UnspecifiedSize && b.currentStatementSize >= b.statementSizeLimit)
 }
 
+// SpeedLimiter used for control speed of dump data
+type SpeedLimiter struct {
+	limiter *rate.Limiter
+	limit  uint64
+}
+
+// CheckSpeed check current speed of dump, if already over, automatically suspended.
+func (sl *SpeedLimiter) CheckSpeed(pCtx *tcontext.Context, newSize uint64) {
+	// Less than or equal to 0 no speed limit.
+	if sl.limit <= 0 {
+		return
+	}
```

Comment:
This if condition is checked everytime. Maybe we can use interface or function variable to avoid this problem?

Reply:
Use interface fix.